shared/test/pageStore_test.js

```diff
@@ -181,8 +220,104 @@ describe("loop.store.PageStore", () => {
     });
   });
 
+  describe("fetchPageMetadata", () => {
+    let actionData;
+    let response;
+
+    beforeEach(() => {
+      actionData = {
+        pageId: "fakeId",
+        url: "http://fakeurl.com",
+        title: "faketitle",
+        thumbnail_img: "fakethumb"
+      };
+
+      response = {
+        title: "fake title",
+        thumbnail_img: "fake thumb"
+      };
+
+      sandbox.stub(loop.shared.utils, "getPageMetadata", url => {
+        if (url === actionData.url) {
+          return Promise.resolve(response);
+        }
+
+        return {
+          then: function() {
+            return {
+              catch: function(cb) {
+                cb();
+              }
+            };
+          }
+        };
+      });
+    });
+
+    it("should dispatch UpdatePage", () => {
+      store.fetchPageMetadata(actionData);
+
+      sinon.assert.calledOnce(dispatcher.dispatch);
+    });
+
+    it("should dispatch UpdatePage with results", () => {
+      store.fetchPageMetadata(actionData);
+
+      sinon.assert.calledWithExactly(dispatcher.dispatch,
+        new actions.UpdatePage({
+          pageId: actionData.pageId,
+          thumbnail_img: response.thumbnail_img,
+          title: response.title
+        }));
+    });
```

Comment:
I think it helps make the tests more easy to debug when the "calledOnce" assertion is separate from the "calledWith" assertion because if the second one fails you can quickly see a green line saying that indeed your stub was called so the problem is with the arguments received.
However I don't feel very strongly about this, I'll update the tests.